src/hotspot/share/gc/g1/heapRegion.cpp

```diff
@@ -138,7 +139,7 @@ void HeapRegion::hr_clear(bool clear_space) {
   init_top_at_mark_start();
   if (clear_space) clear(SpaceDecorator::Mangle);
 
-  _evacuation_failed = false;
+  Atomic::store(&_evacuation_failed, false);
```

Comment:
Can more than one thread attempt this for a given region? I wonder if there really is a need for using `Atomic::store`?